tests/conftest.py

```diff
@@ -54,6 +50,22 @@ def pytest_addoption(parser):
     )
 
 
+@pytest.fixture
```

Comment:
* Caching tests is kind of risky. This is meant only for fixtures correct?
* How many of those do we have and how much time is saved with the cache? I only see the linear shapley values. Are there more?
* If there were more, wouldn't it be better to have a `@pytest.cached_fixture` decorator that acted as a memoizator? I.e. it hashes the inputs, maybe the pip versions, or whatever and so on.

Reply:
- Yes, it is only for test fixtures. Actually there is already such a feature in pytest but it uses json for serialization so I couldn't really use it.
- We only use it for one fixture for now and it divides the test runtime by half.
- I didn't want to optimize it too much because I wasn't sure how it would actually be used and whether we would keep it.

Reply:
By half! Holy 🐮! Ok, maybe then clearly document that it's only meant for fixtures, and if possible add some warning / error / whatever? To deter people from using it on actual tests, or to make them (us) aware of the intended purpose.

Reply:
The main use of pytest's cache is to allow using the `--failed-first` flag to start a test run with the last failed tests.

It says in the documentation that other extension and plugins may access the cache but the fact that it only supports json serialization probably means that it shouldn't be used the way that we're using it.

Still, since for now it is actually saving us some time and it is only used in one single location we can keep it.

See the documentation for more information: https://docs.pytest.org/en/7.1.x/how-to/cache.html

Reply:
Can you document this where it is used? It's going to be hard to remember this in half a year